### ISSUE TITLE
Missing unused import warning for class when a field with the same name is used

### ISSUE THREAD

**Issue by Gama11** [pinterest/ktlint#954]
In this example, the second import is unused, but not reported by ktlint:

`A.kt`:

```kotlin
import foo.Bar.Companion.Bar
import foo.Foo

fun foo() {
    println(Bar.Foo)
}
```

`foo/B.kt`:

```kotlin
package foo

class Foo

class Bar {
    companion object {
        val Bar = Bar()
    }

    val Foo: Int = 0
}
```

This simplified example has the same issue:

```kotlin
import foo.Foo

val Foo = 1

fun foo() {
    println(Foo)
}
```

Possibly related to #563.

## Your Environment

* Version of ktlint used: 0.39.0
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): command line
* Operating System and version: Windows 10

**Comment by paul-dingemans** [comment]
I have been playing around with this but it turns out to be quite complicated. I have modified your example to illustrate the problem that I see:
```
import bar
import foo.Foo1
import foo.Foo2

val Foo1 = 1

fun foo1() {
    println(Foo1) // This prints global variable Foo1 and should not be seen as a justification to import "foo.Foo1" 

    val Foo2 = 2
    println(Foo2) // This prints local variable Foo2 and should not be seen as a justification to import "foo.Foo2" 
}

fun foo2() {
    println(Foo2) // This function has no access a variable called "Foo2". Therefore it is a valid justification to import "foo.Foo2"
}
```

Import statements are checked against a list of reasons why they should be seen as "used" versus "unused". The assignment statements of the global variable `Foo1 = 1` and local variable `Foo2 = 2` are currently of no importance in this evaluation. The reference expressions `Foo1` and `Foo2` in the `println(...)` statements are important in the evaluation. The example above demonstrates that to 'fix' this bug, we would need to evaluate the context in which `Foo1` and `Foo2` are used. This can become very difficult and possibly time consuming.

Given above, I would prefer ktlint to be conservative and not complaining about an import statement that actually is not used. Aggressively removing an import statement that actually is required is not acceptable in my opinion.

**Comment by paul-dingemans** [comment]
> Given above, I would prefer ktlint to be conservative and not complaining about an import statement that actually is not used. Aggressively removing an import statement that actually is required is not acceptable in my opinion.

Closing issue.
